app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -257,6 +264,37 @@ class NavigationOverlayFragment : Fragment() {
         activity!!.moveTaskToBack(true)
     }
 
+    private fun observeAccountState(): Disposable {
+        val fxaRepo = serviceLocator.fxaRepo
+
+        return fxaRepo.accountState.subscribe { accountState ->
+            when (accountState) {
+                is AccountState.AuthenticatedWithProfile -> {
+                    val fxaProfile = accountState.profile
+                    if (fxaProfile.avatar != null) {
+                        PicassoWrapper.client
+                            .load(fxaProfile.avatar.url)
+                            .into(fxaButton, object : Callback {
+                                override fun onError(e: Exception?) {
+                                    Sentry.capture(e)
+                                }
+                                override fun onSuccess() {}
+                            })
+                    }
+                }
+                AccountState.AuthenticatedNoProfile -> {
+                    fxaButton.setImageResource(R.drawable.ic_avatar_authenticated_no_picture)
+                    fxaButton.contentDescription = resources.getString(R.string.fxa_navigation_item_signed_in2)
+                }
+                AccountState.NeedsReauthentication,
+                AccountState.NotAuthenticated -> {
```

Comment:
nit: put these on the same line or put a new line above these two states. Otherwise, it's easy to not notice that both of these states apply to the lambda below.